stetho/src/main/java/com/facebook/stetho/inspector/elements/android/AndroidDocumentProvider.java

```diff
@@ -138,11 +141,16 @@ public NodeDescriptor getNodeDescriptor(Object element) {
   public void highlightElement(Object element, int color) {
     verifyThreadAccess();
 
-    View highlightingView = getHighlightingView(element);
-    if (highlightingView == null) {
+    HighlightableDescriptor highlightableDescriptor = getHighlightableDescriptor(element);
+    if (highlightableDescriptor == null) {
       mHighlighter.clearHighlight();
     } else {
-      mHighlighter.setHighlightedView(highlightingView, color);
+      mHighlightBoundsRect.setEmpty();
+      highlightableDescriptor.getBoundsForHighlighting(element, mHighlightBoundsRect);
```

Comment:
Plus, it looks like `getViewForHighlighting` gets called twice -- once within `getHighlightableDescriptor` and then again when we call `setHighlightedView`. That could hurt performance depending on the implementation.